src/Features/CSharp/Portable/GenerateConstructorFromMembers/CSharpGenerateConstructorFromMembersCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.GenerateConstructorFromMembers;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PickMembers;
+
+namespace Microsoft.CodeAnalysis.CSharp.GenerateConstructorFromMembers
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.GenerateConstructorFromMembers), Shared]
+    [ExtensionOrder(Before = PredefinedCodeRefactoringProviderNames.GenerateEqualsAndGetHashCodeFromMembers)]
+    internal sealed class CSharpGenerateConstructorFromMembersCodeRefactoringProvider
+        : AbstractGenerateConstructorFromMembersCodeRefactoringProvider
+    {
+        [ImportingConstructor]
+        public CSharpGenerateConstructorFromMembersCodeRefactoringProvider()
+        {
+        }
```

Comment:
not needed.

Reply:
Is it the attribute or the constructor that is not needed? Because there's no public parameterless constructor if this one is removed.

Reply:
➡️ Explicit importing constructors are required for all MEF parts. We have analyzer enforcement for this.

Reply:
> Because there's no public parameterless constructor if this one is removed.

Sorry, i missed that tehre was the existin test constructor.  What you did was correct here.